common/utils.cpp

```diff
@@ -246,12 +245,20 @@ GetSubTreeResponse
                             const std::vector<std::string>& ifaceList) const
 {
     auto& bus = pldm::utils::DBusHandler::getBus();
-    auto method = bus.new_method_call(mapperBusName, mapperPath,
-                                      mapperInterface, "GetSubTree");
-    method.append(searchPath, depth, ifaceList);
-    auto reply = bus.call(method);
-    GetSubTreeResponse response;
-    reply.read(response);
+    GetSubTreeResponse response{};
+    try
+    {
+        auto method = bus.new_method_call(mapperBusName, mapperPath,
+                                          mapperInterface, "GetSubTree");
+        method.append(searchPath, depth, ifaceList);
+        auto reply = bus.call(method);
+        reply.read(response);
+    }
+    catch (const std::exception& e)
+    {
+        std::cerr << "failed GetSubTree query for interface " << ifaceList[0]
+                  << " with ERROR=" << e.what() << "\n";
+    }
```

Comment:
no, this is not the right place to catch the exception, any function who is call this getSubtree should have the function call in try - and - catch block. so that at the the function call - we can decide what to do with the generated exception.

Reply:
GetSubTreeResponse response{} is an empty vector. For any exception we will log an error and return empty vector. The function which calls getSubtree will to traverse the vector and takes action on the data present in the vector. So catching the exception here in the util function should be fine.

Reply:
what would happen to this case where , the client code that will call this function need to know if the dbus call failed vs the dbus call returned 0 dbus objects ? and take the necessary action based on it ? 

Right now with this code, the client will always receive {} when the call failed or even when the number of objects found by mapper are 0.

Reply:
I think in both the cases action remains the same. Do you have any specific case were you think this should be handled seperately

Reply:
i don't have any use cases in mind, but I think the community would not agree with this way of designing, we should always give the client an option to choose how they want to handle the generated exceptions, rather than deciding them in the utility function.

Reply:
For now, handling the exception in util function only as we are not taking any specific action on failure in caller function.